src/nimble.nim

```diff
@@ -710,25 +710,57 @@ proc getPackageByPattern(pattern: string, options: Options): PackageInfo =
       )
     result = getPkgInfoFromFile(skeletonInfo.myPath, options)
 
-proc dump(options: Options) =
+# import std/jsonutils
+proc `%`(a: Version): JsonNode = %a.string
+
+proc dump(options: Options, json: bool) =
   cli.setSuppressMessages(true)
   let p = getPackageByPattern(options.action.projName, options)
-  echo "name: ", p.name.escape
-  echo "version: ", p.version.escape
-  echo "author: ", p.author.escape
-  echo "desc: ", p.description.escape
-  echo "license: ", p.license.escape
-  echo "skipDirs: ", p.skipDirs.join(", ").escape
-  echo "skipFiles: ", p.skipFiles.join(", ").escape
-  echo "skipExt: ", p.skipExt.join(", ").escape
-  echo "installDirs: ", p.installDirs.join(", ").escape
-  echo "installFiles: ", p.installFiles.join(", ").escape
-  echo "installExt: ", p.installExt.join(", ").escape
-  echo "requires: ", p.requires.join(", ").escape
-  echo "bin: ", p.bin.join(", ").escape
-  echo "binDir: ", p.binDir.escape
-  echo "srcDir: ", p.srcDir.escape
-  echo "backend: ", p.backend.escape
+  var j: JsonNode
+  var s: string
+  if json: j = newJObject()
+  template fn(key, val) =
+    if json:
+      when val is seq[PkgTuple]:
+        # jsonutils.toJson would work but is only available since 1.3.5, so we
+        # do it manually.
+        j[key] = newJArray()
+        for (name, ver) in val:
+          j[key].add %{
+            "name": % name,
+            # we serialize both: `ver` may be more convenient for tooling
+            # (no parsing needed); while verStr is more familiar.
+            "verStr": % $ver,
```

Comment:
Nit: put this inside the "ver" object as a simple "str" field?

Reply:
renamed verStr to str but kept it out of `ver` which makes more sense IMO; I'm just serializing `VersionRange` in 2 ways: as a human readable string (after `$`), and as an object via `%*` / toJson; moving `str` inside would make things just a little more complicated rather than simpler